modules/programs/taskwarrior.nix

```
@@ -99,5 +99,9 @@ in {
 
       ${cfg.extraConfig}
     '';
+    home.activation.regenDotTaskRc = hm.dag.entryAfter [ "writeBoundary" ] ''
+      $VERBOSE_ECHO "Copying taskwarrior taskrc"
+      $DRY_RUN_CMD echo "include ${xdg.configHome}/task/taskrc_" > ~/.taskrc
```

This would overwrite the file every build/activation.

---

What would be the right way to do this then? I am not that familiar with the code base and the documentation didn't offer an alternative (or I didn't find it).

---

We could grep for the line in the file, if it does not exist we append/prepend the `include` line.

Do you know whether settings specified later in a configuration file override previous settings? If it does, the include line should be the first line of the file.

---

Yup, precedence matters:
```
[2022-02-23 19:50:10] 0 x10an14@home-desktop:~/.config/task
-> $ task context list

Name     Type  Definition                                       Active
test     read  +bar                                             no
         write                                                  no
work     read  +work                                            no
         write +work                                            no

[2022-02-23 19:50:16] 0 x10an14@home-desktop:~/.config/task
-> $ bat taskrc
───────┬───────────────────────────────────────────────────────────────
       │ File: taskrc
───────┼───────────────────────────────────────────────────────────────
   1   │ include /nix/store/485ccd3rgw8qwbrws6w414zcj1dwwx6c-home-manager-files/.config/task/taskrc
   2   │
   3   │ context.test.read=+bar
   4   │
───────┴───────────────────────────────────────────────────────────────
[2022-02-23 19:50:26] 0 x10an14@home-desktop:~/.config/task
-> $ vim taskrc
[2022-02-23 19:50:33] 0 x10an14@home-desktop:~/.config/task
-> $ task context list

Name     Type  Definition                                       Active
test     read  +foo                                             no
         write                                                  no
work     read  +work                                            no
         write +work                                            no

[2022-02-23 19:50:35] 0 x10an14@home-desktop:~/.config/task
-> $ bat taskrc
───────┬───────────────────────────────────────────────────────────────
       │ File: taskrc
───────┼───────────────────────────────────────────────────────────────
   1   │
   2   │ context.test.read=+bar
   3   │ include /nix/store/485ccd3rgw8qwbrws6w414zcj1dwwx6c-home-manager-files/.config/task/taskrc
   4   │
───────┴───────────────────────────────────────────────────────────────
[2022-02-23 19:50:52] 0 x10an14@home-desktop:~/.config/task
-> $

```

---

```suggestion
      $DRY_RUN_CMD ${pkgs.gnused}/bin/sed -i "include ${xdg.configHome}/task/taskrc_" > ~/.taskrc
```

Should do the trick.

---

Won't this script insert the line on every activation?
``` nix
let
  generatedConf = "${config.xdg.configHome}/task/taskrc_";
  userConf = "${config.xdg.configHome}/task/taskrc";
in ''
  grep -qF ${configFile} ${userConfig} || sed -i '1i include ${generatedConf}' ${userConf}
''
```

---

> Won't this script insert the line on every activation?
 
 You're absolutely correct, I forgot to account for that in my previous suggestion >.<